tests/nightly/test_large_array.py

```diff
@@ -1350,6 +1351,50 @@ def run_trsm(inp):
     check_batch_trsm()
 
 
+def test_linalg_large_dim():
+    def check_gemm():
+        A = nd.ones(shape=(1, INT32_MAX + 1, 1))
```

Comment:
this should go to test_large_vector since the input contains 1 dimension which has large while rest dimensions are small. @access2rohit plz confirm

Reply:
Name of the file can be made better.
Basically the idea was to have 2 separate files

1. test_large_array.py [more like large size]
testing input whose individual dimensions are less than `2**32` but size of the input is > `2**32`

2. test_large_vector.py [more like large "shape"]
testing input whose atleast 1 individual dimensions is > `2**32`

Reply:
Maybe we should make more explicit comments on what those test do? I can do this in my next commit. I do think the two cases still both fall into the same category which is testing with tensors of large dimensions.

Reply:
In comments I would say like 1. those tests are for overflowing total size 2. those other tests are for overflowing index calculation i.e. row dim, col dim, etc

Reply:
From consistency standpoint, I'd 
1. put these testss in test_large_vector.py file. 
2. Rename that file to [whatever sounds right I just gave a suggestion above]
3. add a comment in that file.

All dimensions in test_large_array.py are less than INT32_MAX
Large dimension [>`2**32`] was introduced in test_large_vector.py for the same reason.

So to keep the testing approach consistent I'd do that. 

Even if both files play with "Large tensors" one does it for large "size" other specifically for large "shape".

Reply:
well one thing to note is that they are not vectors per se. The inputs are all 3D where as in test_large_vector they are all 1D. The dim checks happen on both row and col dim so you can see I used both (1, 1, x) and (1, x, 1). 

I will add more comments in next commit

Reply:
Yes I know they aren't "vectors" and hence recommended "renaming the file"

Reply:
If we keep these tests in this file, it defeats the purpose of few tests in test_large_vector.py

https://github.com/apache/incubator-mxnet/blob/6bbd53107aa16fc41e8d462cf5dc46fb70d592df/tests/nightly/test_large_vector.py#L99-L126

Reply:
@Zha0q1 We vector file generally houses tests for operators with a single dimension that exceeds 2^32 range. Kindly address what is suggested by @ChaiBapchya and move it to vector tests.

Reply:
Yeah that would make sense. I have moved the tests to test_large_vector.py, which I kept the original name to avoid naming discrepancy with master. Also comments were added to the tests